thoth/storages/graph/postgres.py

```diff
@@ -3974,13 +3975,18 @@ def sync_analysis_result(self, document: dict) -> None:
         with self._session_scope() as session, session.begin(subtransactions=True):
             if is_external:
                 sw_class = ExternalSoftwareEnvironment
+                python_version = None
             else:
                 sw_class = SoftwareEnvironment
+                python_version = image_name.rsplit("-", maxsplit=1)[1]   # pyXX
+                python_version = python_version[2:3] + "." + python_version[3:]   # take first digit and put . after it
+                if not re.match(r"^\d\.\d*$", python_version):
```

Comment:
I think the star should not be present:

```python
>>> import re
>>> re.match(r"^\d\.\d*$", "3.")
<re.Match object; span=(0, 2), match='3.'>
>>> re.match(r"^\d\.\d*$", "3.14159265359")
<re.Match object; span=(0, 13), match='3.14159265359'>
>>> re.match(r"^\d\.\d$", "3.14159265359")
>>> re.match(r"^\d\.\d$", "3.")
>>>
```

Reply:
I don't understand why not?  Isn't it possible to have a python version `3.14149`?  The other option is to have something more like this:
```python
>>> import re
>>> re.match(r"^\d(\.\d+)*$", "3.")
>>> re.match(r"^\d(\.\d+)*$", "3")
<re.Match object; span=(0, 1), match='3'>
```

Reply:
Agh, I missed this one. The issue is with "3." - we require a minor version to be present.